pkg/handlers/catch_all.go

```diff
@@ -0,0 +1,22 @@
+package handlers
+
+import (
+	"fmt"
+	"net/http"
+
+	"go.uber.org/zap"
+)
+
+// CatchAllHandler returns 404
+type CatchAllHandler struct {
+	Logger *zap.Logger
+}
+
+// Handle returns 404 on unexpected routes
+func (h CatchAllHandler) Handle() http.HandlerFunc {
```

Comment:
For coverage, and if we handle this differently in the future, can you add a quick handler test that just ensures this returns a 404.